### PR TITLE
Python eclass tidy-up

### PR DESCRIPTION
**Pull request by mjeveritt**
This is likely to need fixes, but posting it up for comment.
Ping @mgorny .

### CODE REVIEW

File: dev-libs/OpenNI2/OpenNI2-9999.ebuild [hunk 1]
```diff
@@ -32,12 +33,12 @@ RDEPEND="
 	virtual/libudev
 	virtual/jpeg:0
 	opengl? ( media-libs/freeglut )
-	java? ( >=virtual/jre-1.5:* )
+	java? ( >=virtual/jre-1.5:= )
```
Comment (aballier) [inline dev-libs/OpenNI2/OpenNI2-9999.ebuild]: Why?

File: dev-libs/OpenNI2/OpenNI2-9999.ebuild [hunk 2]
```diff
@@ -32,12 +33,12 @@ RDEPEND="
...
 "
 DEPEND="${RDEPEND}
-	dev-lang/python
+	${PYTHON_DEPS}
 	doc? ( app-doc/doxygen )
-	java? ( >=virtual/jdk-1.5:* )"
```
Comment (aballier) [inline dev-libs/OpenNI2/OpenNI2-9999.ebuild]: this must stay...

File: dev-libs/OpenNI2/OpenNI2-9999.ebuild (hunk 1 above)
Comment (mjeveritt) [inline dev-libs/OpenNI2/OpenNI2-9999.ebuild]: Fixed. General ignorance :)

File: dev-libs/OpenNI2/OpenNI2-9999.ebuild (hunk 2 above)
Comment (mjeveritt) [inline dev-libs/OpenNI2/OpenNI2-9999.ebuild]: Reinstated, suspected it would be covered by DEPEND=${RDEPEND} but who am I to question your logic :D

Comment (aballier) [inline dev-libs/OpenNI2/OpenNI2-9999.ebuild]: jdk != jre :)

File: dev-libs/OpenNI2/OpenNI2-2.2_beta2-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE="doc java neon opengl static-libs"
+
+RDEPEND="
+	virtual/libusb:1
+	virtual/libudev
+	virtual/jpeg:0
+	opengl? ( media-libs/freeglut )
+	java? ( >=virtual/jre-1.5:= )
```
Comment (aballier) [inline dev-libs/OpenNI2/OpenNI2-2.2_beta2-r1.ebuild]: seems you didnt update this one after fixing live ebuild

anyway, we should avoid the revbump here, it is a build fix only

File: dev-libs/OpenNI2/OpenNI2-9999.ebuild (hunk 2 above)
Comment (mjeveritt) [inline dev-libs/OpenNI2/OpenNI2-9999.ebuild]: Oh. Class A1 facepalm there .. cheers! :+1:

File: dev-libs/OpenNI2/OpenNI2-2.2_beta2-r1.ebuild (hunk 3 above)
Comment (mjeveritt) [inline dev-libs/OpenNI2/OpenNI2-2.2_beta2-r1.ebuild]: Can do. No problem.

**Comment by mjeveritt** [comment]
OK fully unfudged dev-libs/OpenNI2 @aballier :)

**Comment by aballier** [comment]
Thanks, this looks good to me

@mgorny poked me the other day about PYTHON_REQUIRED_USE, not sure if that'd apply here, let's get his input

File: dev-embedded/libftdi/libftdi-1.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit cmake-utils python-single-r1
+
+MY_P="${PN}1-${PV}"
+SRC_URI="http://www.intra2net.com/en/developer/${PN}/download/${MY_P}.tar.bz2"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~sparc ~x86"
+
+DESCRIPTION="Userspace access to FTDI USB interface chips"
+HOMEPAGE="http://www.intra2net.com/en/developer/libftdi/"
```
Comment (SoapGentoo) [inline dev-embedded/libftdi/libftdi-1.3.ebuild]: drop the libftdi stuff - unless you want to version bump to 1.3, but use mgorny's ebuilds as a basis then, this one's style is too old

**Pull request closed without merging**